js/okx.js

```diff
@@ -4743,46 +4743,40 @@ module.exports = class okx extends Exchange {
         // WARNING: THIS WILL INCREASE LIQUIDATION PRICE FOR OPEN ISOLATED LONG POSITIONS
         // AND DECREASE LIQUIDATION PRICE FOR OPEN ISOLATED SHORT POSITIONS
         if ((leverage < 1) || (leverage > 125)) {
-            throw new BadRequest (this.id + ' setLeverage() leverage should be between 1 and 125');
+            throw new NotSupported (this.id + ' setLeverage () leverage should be between 1 and 125');
         }
         await this.loadMarkets ();
         const market = this.market (symbol);
-        let marginMode = undefined;
-        [ marginMode, params ] = this.handleMarginModeAndParams ('setLeverage', params);
-        if (marginMode === undefined) {
-            marginMode = this.safeString (params, 'mgnMode', 'cross'); // cross as default marginMode
-        }
+        const side = this.safeStringLower2 (params, 'side', 'posSide');
+        params = this.omit (params, [ 'side' ]);
+        const [ marginMode, query ] = this.handleMarginModeAndParams ('setLeverage', params);
         if ((marginMode !== 'cross') && (marginMode !== 'isolated')) {
-            throw new BadRequest (this.id + ' setLeverage() requires a marginMode parameter that must be either cross or isolated');
+            throw new ArgumentsRequired (this.id + ' setLeverage () params["marginMode"] must be either cross or isolated');
+        } else if (marginMode === 'isolated') {
+            if ((side !== 'long') && (side !== 'short')) {
+                throw new ArgumentsRequired (this.id + ' setLeverage () params["side"] must be either "long" or "short"');
+            }
         }
         const request = {
             'lever': leverage,
             'mgnMode': marginMode,
             'instId': market['id'],
+            'posSide': side,
         };
-        const posSide = this.safeString (params, 'posSide');
```

Comment:
We shouldn't do checks for exchange specific params, it gets in the way of users doing specific things with them